pkg/controllers/nodeclaim/termination/suite_test.go

```diff
@@ -175,26 +214,12 @@ var _ = Describe("Termination", func() {
 
 		Expect(env.Client.Delete(ctx, nodeClaim)).To(Succeed())
 		ExpectObjectReconciled(ctx, env.Client, nodeClaimTerminationController, nodeClaim) // triggers the node deletion
-		ExpectObjectReconciled(ctx, env.Client, nodeClaimTerminationController, nodeClaim) // the node still hasn't been deleted, so the nodeClaim should remain
-
-		ExpectExists(ctx, env.Client, nodeClaim)
+		ExpectExists(ctx, env.Client, nodeClaim)                                           // the node still hasn't been deleted, so the nodeClaim should remain
 		ExpectExists(ctx, env.Client, node)
-
 		ExpectFinalizersRemoved(ctx, env.Client, node)
-		ExpectNotFound(ctx, env.Client, node)
-		ExpectObjectReconciled(ctx, env.Client, nodeClaimTerminationController, nodeClaim) // now the nodeClaim should be gone
 
-		ExpectNotFound(ctx, env.Client, nodeClaim)
-	})
-	It("should not call Delete() on the CloudProvider if the NodeClaim hasn't been launched yet", func() {
```

Comment:
Why did we remove this? Do we think that this test no longer applies? How do we handle scenarios where we have no provider id? Seems like we should test this

Reply:
This test no longer applies as per the implementation at least. I had removed that check in one of the previous commits after our discussion to make delete call at least once during the nodeClaim termination flow. I may have misunderstood but if we want to add this test back, we should also add that check in the termination controller flow.